apps/charge-api-service/src/bundler-api/bundler-api.interceptor.ts

```diff
@@ -94,14 +101,72 @@ export class BundlerApiInterceptor implements NestInterceptor {
     return requestConfig
   }
 
-  private prepareUrl (environment) {
+  private prepareUrl (environment, bundlerProvider) {
     if (isEmpty(environment)) throw new InternalServerErrorException('Bundler environment is missing')
-    const config = this.configService.get(`bundler.${environment}`)
+    const config = this.configService.get(`bundler.${bundlerProvider}.${environment}`)
 
     if (config.url) {
       return config.url
     } else {
       throw new InternalServerErrorException(`${capitalize(environment)} bundler environment is missing`)
     }
   }
+
+  private contructUserOp (context: ExecutionContext, requestConfig: AxiosRequestConfig, response) {
+    const request = context.switchToHttp().getRequest()
+    const bundlerProvider = request.query?.provider ?? BundlerProvider.ETHERSPOT
+    const param = requestConfig.data.params[0]
```

Comment:
– The interceptor assumes requestConfig.data.params[0] exists. If a request body or params array is missing, this will throw an error. Add check for its existence

Reply:
@NikolayRn added